### ISSUE TITLE
Bug: `TrackedRecordCollection.Remove()` and `Clear()` offer backdoor to `enforceBusinessLogic`

### ISSUE THREAD

**Issue by JeremyCaney** [Ignia/Topics-Library#79]
Currently, attempting to insert or set a `TrackedRecord<T>` ensures that any business logic is enforced. This is not done, however, when calling `Remove()` or `Clear()`. This is important not only because the business logic might prevent a value being reset to `null`, but also because removing a value might require updating the state of the `Topic`. This is a rare scenario, and it's unlikely to be a major issue—but it should be easy to fix with the `TopicPropertyDispatcher`, and there's no reason not to lock this backdoor.

**Comment by JeremyCaney** [comment]
The challenge with this is that `enforceBusinessLogic` will route through the property, which will then call `SetValue()`—not `Remove()` or `Clear()`. Under some circumstances, `SetValue()` will call `Remove()`, but in others it will update the `TrackedRecord<>` with a null `Value`. In the latter scenario, this yields a confusing result since the keyed item will still exist in the collection. 

### Options

- Keep it as is, acknowledging a potential back door. 
- Update it to `enforceBusinessLogic`, acknowledging potential confusion, if not bugs. 
- Remove `Remove()` and `Clear()`—a breaking change.
- Always remove records, instead of setting them to have a `null` value; rely exclusively on `DeletedItems` for tracking removals. 
- Extend `TopicDispatcher` to include a callback that forces a return to the original calling method. 

> _Note:_ We’ve moved away from subclassing topics and creating property accessors in favor of view model mapping. Given this, it doesn’t make much sense to invest too heavily into this functionality.

**Comment by JeremyCaney** [comment]
Investigating this further, just ensuring nulls are always treated as deletes is one of the easiest option, resolves the underlying issue, and avoids confusion. It’s a good long-term solution without introducing breaking changes or adding a lot of complexity. 

Since business logic is exclusively handled via the `TopicDispatcher.Enforce()`, and not by `SetValue()`, this is relatively easy to do. If the value is null, we just call `Remove()` within `SetValue()`, and then apply the `Enforce()` logic within `RemoveItem()`.

For efficiency, we’ll likely want to refactor the code slightly to prevent the need to create or update a `TrackedRecord<>` in this scenario. For instance, we might just handle this as an escape clause at the top. `TopicDispatcher` already supports `null` values. 

Finally, while we’re at it, we’ll want to remedy an issue where the `!enforceBusinessLogic` bypass gets applied if the `TrackedRecord<>` is null—and, thus, no further action is being taken—as that will prevent the business logic from being applied the _next_ time that key is updated. That can be done by returning in that scenario prior to the call to `Register()`.

**Comment by JeremyCaney** [comment]
To mitigate this, the `SetValue()` logic was updated to call `Remove()` if the `Value` was null (3d1ad6bc), `Remove()` was updated to enforce business logic (0bccf1f3), and `Clear()` was updated to call `Remove()` (abfe213f)—thus enforcing business logic on each individual item. In addition to updating existing tests which assumed that `SetValue()` wouldn't delete a record (7aae9bc0), I also introduced a _new_ unit test to validate that `Clear()` (and, thus, `Remove()`) will, in fact, enforce business logic (fa08bfb2). These updates were merged to `develop` as part of d936bcd.
